al_aws_collector.js

```diff
@@ -147,14 +151,31 @@ class AlAwsCollector {
                 return response.send(event, context, response.FAILED, {Error: exception});
             });
     }
-    
-    checkin(event, context, callback) {
-        // TODO: add stats, etc
+
+    checkin(callback) {
         var collector = this;
-        var checks = this._customHealthChecks;
-        m_healthChecks.getHealthStatus(event, context, checks, function(status) {
-            const checkinValues = Object.assign(collector._getAttrs(), status);
-            collector._azcollectc.checkin(checkinValues)
+        const context = this._invokeContext;
+        const checks = this._customHealthChecks;
+        const statsFuns = this._customStatsFuns;
+
+        //it is assumed that all functions here always return err != null
+        async.parallel([
+            function(asyncCallback) {
+                m_healthChecks.getHealthStatus(context, checks, function(err, healthStatus) {
+                    return asyncCallback(null, healthStatus);
```

Comment:
explicitly return `null` here, + assumption (and comment) that it is expected to never return error.

Are you ok with that?

Reply:
@kkuzmin I was thinking to process `err`here explicitly, but it starts to look like "branchy cranberries" so I decided to stick what is now :)